com.unity.uiwidgets/Runtime/ui/compositing.cs

```diff
@@ -141,13 +141,26 @@ public class PhysicalShapeEngineLayer : _EngineLayerWrapper {
     }
 
     public class SceneBuilder : NativeWrapper {
+        private bool isDisposed = false;
```

Comment:
Try to create a new Class, e.g., NativeWrapperCPtrDisposable and putting all the DisposeCPtr relevant logics inside. Then SceneBuilder, Image, etc. can inherit from this class